### DISCUSSION TITLE
Accidentally set mailto handler

### DISCUSSION THREAD

**Discussion by SlidingHorn** [qutebrowser/qutebrowser#6581] [Q&A] [answered]
**Version:** 2.3.0
**Install Method:** via git & venv

I accidentally said "yes" when asked if I wanted to have qutebrowser open all mailto links with an old gmail account, and I cannot seem to find where to undo this.

Any tips?  (asked in the Matrix/IRC too, but it's not very active)

**Comment by mschilli87** [comment]
In what application did you click "yes"? In qutebrowser? If not we might at least need to know you operating system. :wink:
I have never seen a dialog offering me to open `mailto:` links in qutebrowser myself, so it'll be hard to help you figure out how to undo it. Plus, how is your default application to open certain links at all related to your email provider?

**Reply by SlidingHorn** [reply]
Yes, in qutebrowser - it popped up similarly to  the messages like "do you want to accept this site's certificate" etc 

Just in case, OS is Ubuntu 20.04

**Comment by toofar** [comment]
If it is even remembered across restarts it'll be in data/webengine somewhere. I don't think qutebrowser persists anything to do with registrations.

**Comment by The-Compiler** [comment]
> Any tips? (asked in the Matrix/IRC too, but it's not very active)

FWIW I didn't see anything there, but the Matrix bridge had a bit of trouble yesterday I think. I'll answer to the rest in a separate comment.

**Comment by The-Compiler** [comment, marked as answer]
> I accidentally said "yes" when asked if I wanted to have qutebrowser open all mailto links with an old gmail account, and I cannot seem to find where to undo this.

The dialog sets the `content.register_protocol_handler` option. If you answered "yes" and not "always", from qutebrowser's point of view, nothing was persisted. You can check via `:config-diff` to be sure.

I'm not sure if the underlying QtWebEngine/Chromium persists that, though. According to [this post](https://askubuntu.com/questions/465586/how-to-reset-external-protocol-handler-in-chrome) it used to, but doesn't do so anymore. What QtWebEngine version are you on (see `:version`)?

**Reply by SlidingHorn** [reply]
> The dialog sets the `content.register_protocol_handler` option. If you answered "yes" and not "always", from qutebrowser's point of view, nothing was persisted. You can check via `:config-diff` to be sure.

I don't see it set there....is there a difference between "y" and "Y" (i.e. "y" = yes and "Y" = always) in that dialog?

> I'm not sure if the underlying QtWebEngine/Chromium persists that, though. According to [this post](https://askubuntu.com/questions/465586/how-to-reset-external-protocol-handler-in-chrome) it used to, but doesn't do so anymore. What QtWebEngine version are you on (see `:version`)?

Backend: QtWebEngine 5.15.2

**Reply by SlidingHorn** [reply]
okay, well I finally restarted the browser, and nothing (not even my keybindings) are there anymore, so now I gotta go figure that out again :(

This particular issue can be closed though

**Reply by SlidingHorn** [reply]
> okay, well I finally restarted the browser, and nothing (not even my keybindings) are there anymore, so now I gotta go figure that out again

I hadn't been using a config.py - things seemed to persist beforehand regardless, but *shrug*

Now there's a config.py set, so there shouldn't be any further issues on that front.

**Reply by toofar** [reply]
> I don't see it set there....is there a difference between "y" and "Y" (i.e. "y" = yes and "Y" = always) in that dialog?

'y' just sets it in memory, 'Y' will write it to the autoconfig.yml file too

> restarted the browser, and nothing (not even my keybindings) are there anymore
> I hadn't been using a config.py

config.py and autoconfig.yml (what `:set` etc write to) are alternative ways of configuring. You can get both by putting `config.load_autoconfig()` in your config.py to load the stuff save to the yaml file. ref https://qutebrowser.org/doc/help/configuring.html#configpy-autoconfig
